component-library/getting-started/customization.mdx

```diff
@@ -0,0 +1,128 @@
+
+You can pass an Appearance object to an individual component, or pass it in to the `ComponentLibraryProvider` to customize elements globally. Let's look at an example of a Signup component that's customized with the `appearance` prop:
+
+```jsx
+import * as React from "react";
+import { useNavigate } from "react-router-dom";
+import { Signup } from "@propelauth/react";
+
+export default function SignupPage() {
+  const navigate = useNavigate();
+
+  const signupAppearance = {
+    options: {
+      divider: false,
+      submitButtonText: "Enter the Dungeon",
+    },
+    elements: {
+      Container: { backgroundColor: "gray" },
+      SubmitButton: "custom-class",
+    },
+  };
+
+  return (
+    <div>
+      <Signup
+        onSignupCompleted={() => navigate("/dashboard")}
+        onRedirectToLogin={() => navigate("/login")}
+        appearance={signupAppearance}
+      />
+    </div>
+  );
+}
+```
+
+Each component has a unique appearance prop that maps to the options and elements available for that component. For example, here's the full `SignupAppearance` type which shows all options and elements that you can use to customize the [Signup](/component-library/components/signup) component:
+
+```typescript
+type ElementAppearance<T> =
+  | null
+  | string
+  | CSSProperties
+  | (props: T) => JSX.Element;
+
+type SignupAppearance = {
+  options?: {
+    displayLogo?: boolean;
+    divider?: ReactNode | boolean;
+    submitButtonText?: ReactNode;
+  };
+  elements?: {
+    Container?: ElementAppearance<ContainerProps>;
+    Logo?: ElementAppearance<ImageProps>;
+    Header?: ElementAppearance<H3Props>;
+    Divider?: ElementAppearance<DividerProps>;
+    FirstNameLabel?: ElementAppearance<LabelProps>;
+    FirstNameInput?: ElementAppearance<InputProps>;
+    LastNameLabel?: ElementAppearance<LabelProps>;
+    LastNameInput?: ElementAppearance<InputProps>;
+    UsernameLabel?: ElementAppearance<LabelProps>;
+    UsernameInput?: ElementAppearance<InputProps>;
+    EmailLabel?: ElementAppearance<LabelProps>;
+    EmailInput?: ElementAppearance<InputProps>;
+    PasswordLabel?: ElementAppearance<LabelProps>;
+    PasswordInput?: ElementAppearance<InputProps>;
+    SocialButton?: ElementAppearance<ButtonProps>;
+    SubmitButton?: ElementAppearance<ButtonProps>;
+    RedirectToLoginLink?: ElementAppearance<ButtonProps>;
+    RedirectToPasswordlessLoginButton?: ElementAppearance<ButtonProps>;
+    RedirectToSSOLoginButton?: ElementAppearance<ButtonProps>;
+    ErrorMessage?: ElementAppearance<AlertProps>;
+  };
+};
+```
+
+## Customizing elements
+
+You can customize elements in a variety of ways. If the value of an element is a string, that string will be applied as a class name to the element. If it's an object containing CSS properties, those CSS properties will get applied as inline styles. If the value is a React component, that component will override the existing element. Lastly, if the value is null, it won't render the element.
+
```

Comment:
Possibly more of a preference thing, but from a docs point of view it might be nice to break the four ways you can customize elements into subsections and show an example of each, then showing an example that uses all four